tmc-prototype/DishMaster/DishMaster/DishMaster.py

```diff
@@ -0,0 +1,744 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of the DishMaster project
+#
+#
+#
+# Distributed under the terms of the GPL license.
+# See LICENSE.txt for more info.
+
+""" DishMaster Simulator
+
+SKA Dish Master TANGO device server
+"""
+
+# Additional import
+# PROTECTED REGION ID(DishMaster.additionnal_import) ENABLED START #
+import time
+import threading
+from threading import Timer
+
+# tango imports
+import tango
+from tango import DebugIt, DevState, AttrWriteType
+from tango.server import run, DeviceMeta, attribute, command, device_property
+from SKAMaster import SKAMaster
+
+import CONST
+
+# PROTECTED REGION END #    //  DishMaster.additionnal_import
+
+__all__ = ["DishMaster", "main"]
+
+
+class DishMaster(SKAMaster):
+    """
+    SKA Dish Master TANGO device server
+    """
+    __metaclass__ = DeviceMeta
+    # PROTECTED REGION ID(DishMaster.class_variable) ENABLED START #
+
+    # Function to set achieved pointing attribute to the desired pointing attribute
+
+    def point(self):
+        """ This method sets achieved pointing attribute to the desired pointing attribute """
+        if((self._achieved_pointing[1] != self._desired_pointing[1]) |
+           (self._achieved_pointing[2] != self._desired_pointing[2])):
+            try:
+                self.azimuth_thread = threading.Thread(None, self.azimuth, 'DishMaster')
+                self.elevation_thread = threading.Thread(None, self.elevation, 'DishMaster')
+                self.azimuth_thread.start()
+                self.elevation_thread.start()
+                self._pointing_state = 1
+
+            except Exception as except_occured:
+                print CONST.ERR_EXE_POINT_FN, self.ReceptorNumber
+                print CONST.STR_ERR_MSG, except_occured
+
+    def azimuth(self):
+        """ This method calculates the azimuth difference """
+        self._pointing_state = 1
+        self._azimuth_difference = self._desired_pointing[1] - self._achieved_pointing[1]
+        if self._azimuth_difference > 0.0:
+            self.increment_position([1, self._azimuth_difference])
+        elif self._azimuth_difference < 0.0:
+            self.decrement_position([1, abs(self._azimuth_difference)])
+
+    def elevation(self):
+        """ This method calculates the elevation difference """
+        self._pointing_state = 1
+        self._elevation_difference = self._desired_pointing[2] - self._achieved_pointing[2]
+        if self._elevation_difference > 0.0:
```

Comment:
Added extra 0s after decimal point so that the values can be checked as accurately as possible. Commit id is 2225d974d7c44901baea14d30d09a2a978be18db.